crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProviderName.kt

```diff
@@ -0,0 +1,43 @@
+package net.corda.v5.crypto.merkle
+
+import net.corda.v5.base.annotations.CordaSerializable
+
+/**
+ * The hash digest provider name.
+ *
+ * @property name The name of the hash digest provider to be used for the instance.
+ */
+@CordaSerializable
+class MerkleTreeHashDigestProviderName(val name: String) {
+    init {
+        require(name.isNotBlank()) { "Hash digest provider name unavailable or not specified" }
+    }
+
+    companion object {
+        @JvmField
+        val DEFAULT: MerkleTreeHashDigestProviderName = MerkleTreeHashDigestProviderName("DefaultHashDigestProvider")
```

Comment:
needs text to say what algorithm they are. ie.
DEFAULT - An implementation of rfc6962 compatible merkle tree
NONCE - A merkle tree digest that one-way derives a set of per-leaf nonces from a 16-byte entropy input. The nonces are pre-hashed with the leaf data before forming a standard merkle tree. This allows low entropy inputs to be used in the merkle tree.
NONCE_VERIFY - When verifying a NONCE digest based MerkleProof the source entropy is not communicated to the verifier. Instead only the per-leaf nonces are transmitted. Use the NONCE_VERIFY digest to validate the MerkleProof against the expected root hash.
NONCE_SIZE_ONLY_VERIFY - Using the NONCE digest it is possible to create a MerkleProof that doesn't reveal the leaf data, but provides supporting evidence of the number of leaves. These MerkleProofs can be validated against and expected root hash using the NONCE_SIZE_ONLY_VERIFY digest
TWEAKABLE - This provides an extension of the DEFAULT hash with configurable hashes for leaf and internal Merkle node operations. This is required when similar Merkle trees are used to generate a hash that is signed over so that replay attacks can be prevented.